content/6_Advanced/Max_Flow.mdx

```diff
@@ -89,6 +90,133 @@ int main() {
 </CPPSection>
 </LanguageSection>
 
+### Push-Relabel Algorithm
+
+The [Push-Relabel Algorithm](https://en.wikipedia.org/wiki/Push%E2%80%93relabel_maximum_flow_algorithm) is an alternative solution to finding the maximum flow.
+
+To find the maximum flow, we'll handle a **preflow**.
+The only difference between this and a normal flow is that the incoming flow can exceed the out coming flow.
+
+Let's define the excess flow of node u as: $in_u - out_u$, where $in$ is the incoming flow and $out$ is the outgoing flow.
+
+The algorithm starts with an initial preflow where the source has an excess flow. At every stage, it picks a node with the excess flow and tries to **push** the excess to its neighbors, if the capacity supports it.
+To push excess flow from node u to node v we define $\Delta = min(excess_u, capacity_{u,v} - flow_{u,v})$, where $\Delta$ is the maximum supported flow by the
+edge $(u, v)$. The process stops when no more excess nodes are left in the flow network.
+
+Another important feature of the algorithm is the **labeling** function $h$, also known as the height function, which assigns each node an
+integer. One labeling in valid if satisfies the following conditions: $h_{source} = |V|$, $h_{sink} = 0$ and $h_u \le h_v + 1$ if there is an edge from node $u$
+to node $v$ with positive capacity, i.e. it supports more flow. The height function helps us know where to send the excess flow, and where it's needed. It's like water, it can only
+flow from top to bottom.
```

Comment:
```suggestion
Another important feature of the algorithm is the **labeling** function $h$, also known as the height function, which assigns each node an
integer. One labeling in valid if satisfies the following conditions: $h_{source} = |V|$, $h_{sink} = 0$, and $h_u \le h_v + 1$. If there is an edge from node $u$
to node $v$ with positive capacity, i.e. it supports more flow. The height function tells us where to send the excess flow, and where it's needed. It's like water, it can only
flow from top to bottom.
```

(optional)

if u can, please try and reword or split this (specifically the "supports more flow"), it just reads a bit confusing imo